### PR TITLE
web plugin: Improve UI with bootstrap 4

### PR DESCRIPTION
**Pull request by matlads**
These commits do the following: 

1. updated backbone, jquery, 
2. introduces howler (the audio library), and bootstrap css framework
3. uses the /stats, /album, and search API end points
4. introduces the concept of a queue for playing music from the collection
5. updates documentation accordingly

### CODE REVIEW

**Comment by sampsyo** [comment]
Awesome; thank you for keeping up the good work! To confirm, this is a successor to #3079, right? @dbogdanov, did you want to take a look?

**Comment by matlads** [comment]
> Awesome; thank you for keeping up the good work! To confirm, this is a successor to #3079, right? @dbogdanov, did you want to take a look?

Yes it is the successor to #3079.

**Comment by eidoom** [comment]
@matlads This web UI is looking really nice! 

One thing I noticed: everything is working when I run it on http://127.0.0.1:8337/ but when I run it to be accessed over my local network as, say, http://192.168.0.2:8337/ then it is as though my library is empty. Artists, albums and stats are all empty and searches return no results. Is this something you have tested? And do others reproduce this behaviour?

**Comment by jackwilsdon** [comment]
@eidoom have you tried setting `host` to `0.0.0.0`? 

https://beets.readthedocs.io/en/stable/plugins/web.html#configuration

**Comment by eidoom** [comment]
> @eidoom have you tried setting `host` to `0.0.0.0`?
> 
> https://beets.readthedocs.io/en/stable/plugins/web.html#configuration

This gives the same behaviour for me: empty library on the network address and working on the localhost address.

File: beetsplug/web/static/js/beets.js [hunk 1]
```diff
@@ -0,0 +1,562 @@
+/* globals _,$,Backbone,Marionette,Howl */
+(function () {
+    "use strict";
+
+    // Constants
+    var SETTINGS = {
+        API: "http://127.0.0.1:8337",
```
Comment (jackwilsdon) [inline beetsplug/web/static/js/beets.js]: @eidoom looks like this here is the problem. This needs changing before merging @matlads, we should be able to use `window.location.origin` here instead.

```suggestion
        API: window.location.origin,
```

Comment (eidoom) [reply beetsplug/web/static/js/beets.js]: This fixed the problem for me :) Thank you @jackwilsdon

**Comment by dbogdanov** [comment]
This definitely looks much better than before! Works fine for me on localhost. This can be merged to foster further progress on top of it. All ideas for improvements can go to a new feature request issue.

**Pull request closed without merging**